receiver/k8sclusterreceiver/collection/metadata.go

```diff
@@ -46,27 +49,40 @@ type KubernetesMetadata struct {
 // live on v1.ObjectMeta.
 func getGenericMetadata(om *v1.ObjectMeta, resourceType string) *KubernetesMetadata {
 	rType := strings.ToLower(resourceType)
-	metadata := util.MergeStringMaps(map[string]string{}, om.Labels)
+	metadata := util.MergeStringMaps(map[string]string{}, prefixLabels(om.Labels))
 
 	metadata[k8sKeyWorkLoadKind] = resourceType
 	metadata[k8sKeyWorkLoadName] = om.Name
 	metadata[fmt.Sprintf("%s.creation_timestamp",
 		rType)] = om.GetCreationTimestamp().Format(time.RFC3339)
 
 	for _, or := range om.OwnerReferences {
-		metadata[strings.ToLower(or.Kind)] = or.Name
-		metadata[strings.ToLower(or.Kind)+"_uid"] = string(or.UID)
+		kind := strings.ToLower(or.Kind)
+		metadata[getOTelNameFromKind(kind)] = or.Name
+		metadata[getOTelUIDFromKind(kind)] = string(or.UID)
 	}
 
 	return &KubernetesMetadata{
-		resourceIDKey: getResourceIDKey(rType),
+		resourceIDKey: getOTelUIDFromKind(rType),
 		resourceID:    metadataPkg.ResourceID(om.UID),
 		metadata:      metadata,
 	}
 }
 
-func getResourceIDKey(rType string) string {
-	return fmt.Sprintf("k8s.%s.uid", rType)
+func getOTelUIDFromKind(kind string) string {
```

Comment:
unless OTel UIDs are a specific thing, `getPrefixedUIDKey()` or similar conveys more (similarly for OTel names below).

Reply:
These methods derive the OTel conventions listed [here](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/semantic_conventions/k8s.md).